src/mod/transforms_maximal_overlap.jl

```diff
@@ -0,0 +1,102 @@
+
+"""
+Perform one level of the maximal overlap discrete wavelet transform (MODWT) on
+the the vector `v`, which is either the `j`th level scaling coefficients or,
+if `j==1`, the raw signal.  The vectors `h` and `g` are the MODWT detail and
+scaling filters.
+
+Returns a tuple `(v, w)` of the scaling and detail coefficients at level `j+1`.
+"""
+function modwt_step(v::Array{T,1}, j::Integer, h::Array{S,1},
+        g::Array{S,1}) where {T <: Number, S <: Number}
+    N = length(v)
+    L = length(h)
+    v1 = zeros(v)
+    w1 = zeros(v)
+    for t in 1:N
+        k = t
+        w1[t] = h[1] * v[k]
+        v1[t] = g[1] * v[k]
+        for n in 2:L
+            k -= 2^(j-1)
+            if k <= 0
+                k = mod1(k, N)
+            end
+            w1[t] += h[n] * v[k]
+            v1[t] += g[n] * v[k]
+        end
+    end
+    return v1, w1
+end
+
+
+"""
+Perform a maximal overlap discrete wavelet transform (MODWT) of the array
+`x`.  The wavelet type `wt` determines the transform type and the wavelet
+class, see `wavelet`.
+
+The number of transform levels `L` can be any number <= `maxmodwttransformlevels(x)`
+(the default value).
+
+Returns an `n × L+1` matrix (where `n` is the length of `x`) with the wavelet
+coefficients for level j in column j.  The scaling coefficients are in the
+last (L+1th) column.
+"""
+function modwt(x::Array{T,1}, wt::OrthoFilter,
+        L::Integer=maxmodwttransformlevels(x)) where T <: Number
+    @assert L <= maxmodwttransformlevels(x)
+    g, h = WT.makereverseqmfpair(wt)
+    g /= sqrt(2)
+    h /= sqrt(2)
```

Comment:
For the MODWT, the filters are scaled by `1/sqrt(2)` so that the transformed coefficients have the same total energy/variance as the original signal.  Maybe this scaling could be made a default argument to `modwt` so the user can change it if they want to?

Reply:
Ok, or it could be stated in the documentation for the function that the filters are scaled like you said.